### PR TITLE
Refactores the nasty switch statement. 

### PR DESCRIPTION
**Pull request by morloderex**
@marktopper

### CODE REVIEW

**Comment by morloderex** [comment]
A rebase might have been wrong here..

**Comment by fletch3555** [comment]
Fixed your branch.  You had extra commits since `1.x-dev` was out of date with `1.0`.

**Comment by morloderex** [comment]
Hehe how should i know that.. :P

**Comment by fletch3555** [comment]
Not saying you should have, nor that it was your fault.  Just informing you of the change, that's all.

**Comment by emptynick** [comment]
Is there a reason that this is not implemented in the Formfield-class itself?

File: src/Http/Controllers/ContextTypes/BaseType.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace TCG\Voyager\Http\Controllers\ContentTypes;
```
Comment (fletch3555) [inline src/Http/Controllers/ContextTypes/BaseType.php]: You have a typo in all of your new classes.  You used the namespace `ContentTypes`, but named the folder `ContextTypes`, so the classes aren't being found.  This is causing tests to fail.  Please fix it.

**Review by fletch3555 — CHANGES_REQUESTED** [review]

**Comment by morloderex** [comment]
whoops. Sorry @fletch3555 Fixed.

@emptynick This could also be done in that way, but i was trying to keep it the way it was as much as possible. But you could agrue that it should go in there.

File: src/Http/Controllers/ContextTypes/BaseType.php (hunk 1 above)
Comment (morloderex) [reply src/Http/Controllers/ContextTypes/BaseType.php]: This has been fixed

**Review by marktopper — APPROVED** [review]
Great refactor

**Comment by marktopper** [comment]
NOTE: If I remember right, then StyleCI fixes are automatically fixed upon merge.

**Review by fletch3555 — APPROVED** [review]

**Comment by fletch3555** [comment]
Correct @marktopper.  I'm hesitant to merge this at the moment, because I know this whole area of the app is fairly fragile and needs a major refactor (in the works, I promise).  But I'll merge it in.  I don't see this change causing any issues unless there's a significant copy/paste error we didn't see.

**Pull request merged**